Add group role support to superuser whitelist and fix regtests

mpalmi opened the pull request:
This feature should resolve Issue #7.

mpalmi committed:
Add group role support to superuser whitelist

This feature adds support for '+<group_role>' syntax in the superuser
whitelist. This should allow users to be added to the whitelist by
association with the specified "<group_role>".

Resolves: #7

jconway reviewed (commented):
I think we want to use has_privs_of_role() rather than is_member_of_role(). The former does not recurse through roles that don't have rolinherit set. The latter does recurse through roles regardless of rolinherit. See src/backend/utils/adt/acl.c around line 4805 (currently, in pg10 sources).

mpalmi committed:
Fix regression test failure for set_user_u

Re-run the regression test on top of the most recent feature branch and
produce matching results.

Resolves: #9
